proxy/src/signal.rs

```diff
@@ -0,0 +1,81 @@
+//! Unix signal handling for the proxy binary.
+
+extern crate futures;
+extern crate tokio_core;
+
+use self::futures::Future;
+use self::tokio_core::reactor::Handle;
+
+type ShutdownSignal = Box<Future<Item=(), Error=()> + Send>;
+
+/// Returns a `Future` that completes when the proxy should start to shutdown.
+pub fn shutdown(handle: &Handle) -> ShutdownSignal {
+    imp::shutdown(handle)
+}
+
+#[cfg(unix)]
+mod imp {
+    extern crate tokio_signal;
+
+    use std::fmt;
+
+    use self::tokio_signal::unix::{Signal, SIGINT, SIGTERM};
+    use super::futures::{future, Future, Stream};
+    use super::{Handle, ShutdownSignal};
+
+    pub(super) fn shutdown(handle: &Handle) -> ShutdownSignal {
+
+        // If you add to this list, you should also update DisplaySignal below
+        // to output nicer signal names.
```

Comment:
Let's document that SIGTERM is sent by Kubernetes and Docker for graceful shutdown and we support SIGINT to allow simulating that signal easily with Ctrl+C.